### PR TITLE
improve(MultiCallerClient): Correctness & test improvements

### PR DESCRIPTION
**Pull request by pxrl**
- Verify robustness against a 0-length array being supplied to buildMultiCallBundles().
 - Remove async on buildMultiCallBundles().
 - Improve test naming.

### CODE REVIEW

File: src/clients/MultiCallerClient.ts [hunk 1]
```diff
@@ -407,12 +407,13 @@ export class MultiCallerClient {
...
   }
 
-  async buildMultiCallBundles(
+  buildMultiCallBundles(
     txns: AugmentedTransaction[],
     chunkSize = DEFAULT_MULTICALL_CHUNK_SIZE
-  ): Promise<AugmentedTransaction[]> {
-    const txnChunks: AugmentedTransaction[][] = lodash.chunk(txns, chunkSize);
+  ): AugmentedTransaction[] {
+    if (txns.length === 0) return [];
```
Comment (mrice32) [inline src/clients/MultiCallerClient.ts:414]: nit: why is this needed? Does lodash.chunk not handle length === 0?

**Review by mrice32 — APPROVED** [review]
Looks great!

File: src/clients/MultiCallerClient.ts (hunk 1 above)
Comment (pxrl) [reply src/clients/MultiCallerClient.ts:414]: The problem is actually on line 419:

```
return txnChunk.length > 1 ? this.buildMultiCallBundle(txnChunk) : txnChunk[0];
```

If we pass in zero-length array to `buildMultiCallBundles()`, lodash will (correctly) produce the same, and then at line 419 we return element 0 because txnChunks.length is <= 1.

Demonstration:
````
> let y = lodash.chunk([], 50)
undefined
> y
[]
> y[0]
undefined
```

I considered chaining ternary statements on line 419 but it's pretty messy, and I think it's conceptually wrong to make it that far with zero-length input:

```
return txnChunk.length > 1
  ? this.buildMultiCallBundle(txnChunk)
  : txnChunk.length === 1 ? txnChunk[0] : [];
```

**Review by mrice32 — COMMENTED** [review]
https://github.com/across-protocol/relayer-v2/pull/434#discussion_r1085536427

File: src/clients/MultiCallerClient.ts (hunk 1 above)
Comment (pxrl) [reply src/clients/MultiCallerClient.ts]: Schooled once again! Thanks :)
